medmodels/data_synthesis/synthesizer_model.py

```diff
@@ -0,0 +1,87 @@
+"""This module defines the SynthesizerModel class, which is a PyTorch module that
+is used to generate synthetic data using a synthesizer model, after the synthesizer
+model has been trained on real data."""
+
+from abc import ABCMeta, abstractmethod
+from pathlib import Path
+from typing import Any, Union
+
+import sparse
+import torch
+from torch import nn
+
+from medmodels import MedRecord
+from medmodels.data_synthesis.synthesizer import Synthesizer
+
+
+class SynthesizerModel(nn.Module, metaclass=ABCMeta):
+    """SynthesizerModel is an abstract class to define synthesizer models.
+
+    This class should be instantiated once a Synthesizer has been trained on real data.
+    """
+
+    number_of_samples: int
+    device: torch.device
+
+    def __init__(
+        self,
+        synthesizer: Synthesizer,
+        number_of_samples: int,
+        device: torch.device,
+        **kwargs: Any,
```

Comment:
this is not an abstract method, so any unused arguments should be removed

```suggestion
    def __init__(
        self,
        number_of_samples: int,
        device: torch.device,
```

However this might be an actual abstract method but then you'd have to remove the `**kwargs` (same discussion as in synthesizer)